hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerReportHandler.java

```diff
@@ -47,16 +47,17 @@ public class ContainerReportHandler extends AbstractContainerReportHandler
   private static final Logger LOG =
       LoggerFactory.getLogger(ContainerReportHandler.class);
 
-  private final NodeManager nodeManager;
-  private final ContainerManager containerManager;
-  private final String unknownContainerHandleAction;
+  enum UnknownContainerAction {
+    WARN, DELETE;
 
-  /**
-   * The action taken by ContainerReportHandler to handle
-   * unknown containers.
-   */
-  static final String UNKNOWN_CONTAINER_ACTION_WARN = "WARN";
-  static final String UNKNOWN_CONTAINER_ACTION_DELETE = "DELETE";
+    static UnknownContainerAction parse(String s) {
+      return s.equals(WARN.name()) ? WARN
+          : s.equals(DELETE.name()) ? DELETE
+          : null;
+    }
```

Comment:
Should we use built-in `valueOf`?  I think throwing exception on invalid value (which `valueOf` does) is better than silently ignoring it.

Reply:
valueOf will throw IllegalArgumentException.
```
Exception in thread "main" java.lang.IllegalArgumentException: No enum constant org.apache.hadoop.hdds.scm.container.ContainerReportHandler.UnknownContainerAction.abc
	at java.base/java.lang.Enum.valueOf(Enum.java:240)
```

BTW, the original code below will do nothing if the conf is neither WARN nor DELETE.  Let me change it WARN as default.
https://github.com/apache/ozone/blob/30e4aa44d862b0cfc607360bb7f9df1e013955ed/hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/ContainerReportHandler.java#L227-L237